File: server/app/schemas/event.py
from pydantic import BaseModel, Field


class EventChoiceRequest(BaseModel):
    character_id: int = Field(..., gt=0)
    choice_index: int = Field(..., ge=0)


class EventChoiceBrief(BaseModel):
    index: int
    text: str


class PendingEventResponse(BaseModel):
    id: int
    title: str
    description: str
    choices: list[EventChoiceBrief]


class EventChoiceResultResponse(BaseModel):
    narrative: str
    effects: dict
    new_stats: dict
    triggered_events: list

File: client/src/types/event.ts
export interface EventChoiceRequest {
  character_id: number;
  choice_index: number;
}

export interface EventChoiceResult {
  narrative: string;
  effects: Record<string, number>;
  new_stats: { health: number; intelligence: number; charisma: number; wealth: number; happiness: number; luck: number };
  triggered_events: { id: number; title: string; description: string }[];
}
